templates/service-worker.js

```diff
@@ -94,3 +94,123 @@ self.addEventListener('fetch', event => {
     return fetch(req)
   })())
 })
+
+self.addEventListener('push', event => {
+  event.waitUntil((async () => {
+    const data = event.data.json()
+    const { origin } = new URL(data.icon)
```

Comment:
Can't think of a better option to get the instance from the notification payload.

Reply:
This works fine AFAICT1